test/integration/delete.js

```diff
@@ -83,6 +87,28 @@ module.exports = function (createFn, setup, dismantle) {
           done()
         })
       })
+
+      it('DELETE /Customers?query={"name":"John"} 200 - exact match', function (done) {
+        request.del({
+          url: util.format('%s/api/v1/Customers', testUrl),
+          qs: {
+            query: JSON.stringify({
+              name: 'John'
+            })
+          },
+          json: true
+        }, function (err, res, body) {
+          assert.ok(!err)
+          assert.equal(res.statusCode, 204)
+
+          db.models.Customer.find({}, function (err, customers) {
+            assert.equal(customers.length, 2)
+            assert.ok(customers[0].name !== 'John')
```

Comment:
We should iterate with forEach so we don't have to refactor this code if we had Customer objects in the future, same thing for line 204